src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java

```diff
@@ -204,51 +205,86 @@ private void updateJudgement(CaseData caseData, JudgementType judgementType) {
     }
 
-    private void updateRespondentRep(CaseData caseData, RepresentedTypeR representedType) {
+    private void updateRespondentRep(CaseData caseData, UpdateDataModel updateDataModel) {
 
-        //case 1: update is adding
+        if (caseData.getRespondentCollection() != null) {
 
-        //case 2: update is removing
+            RepresentedTypeR representedType = updateDataModel.getRepresentedType();
+            String isRespondentRepRemovalUpdate = updateDataModel.getIsRespondentRepRemovalUpdate();
+            boolean respondentRepRemovalUpdate = isRespondentRepRemovalUpdate.equals(YES);
 
-        if (caseData.getRespondentCollection() != null) {
             Optional<RespondentSumTypeItem> respondentSumTypeItemOptional =
                     caseData.getRespondentCollection().stream()
                             .filter(respondentSumTypeItem ->
                                     respondentSumTypeItem.getValue().getRespondentName()
                                             .equals(representedType.getRespRepName()))
                             .findAny();
 
-            if (respondentSumTypeItemOptional.isPresent()) {
+            if(respondentSumTypeItemOptional.isPresent()){
+                if (!respondentRepRemovalUpdate) {
+                    addRespondentRepUpdates(caseData, updateDataModel);
+                }else {
+                    addRespondentRepRemovalUpdate(caseData, representedType);
+                }
+            }
+        }
+    }
 
-                if (caseData.getRepCollection() != null) {
+    private void addRespondentRepRemovalUpdate(CaseData caseData, RepresentedTypeR representedType){
+        if(representedTypeRItemExists(caseData, representedType)){
+            getExistingRepresentedTypeRItem(caseData, representedType).setValue(new RepresentedTypeR());
+        }
+    }
 
-                    boolean found = false;
+    private void addRespondentRepUpdates(CaseData caseData, UpdateDataModel updateDataModel){
+        boolean caseDataHasRep = (caseData.getRepCollection() != null);
+        RepresentedTypeR representedType = updateDataModel.getRepresentedType();
 
-                    for (RepresentedTypeRItem representedTypeRItem : caseData.getRepCollection()) {
+        if(caseDataHasRep) {
 
-                        if (representedTypeRItem.getValue().getRespRepName()
-                                .equals(representedType.getRespRepName())) {
+            if(representedTypeRItemExists(caseData, representedType)){
+                RepresentedTypeRItem representedTypeRItem = getExistingRepresentedTypeRItem(caseData,
+                        representedType);
+                if(representedTypeRItem != null)
+                {
+                    representedTypeRItem.setValue(representedType);
+                }
+            }
+            else{
+                caseData.getRepCollection().add(createRespondentRepTypeItem(representedType));
+            }
```

Comment:
Could you improve the formatting of this. There are two different styles being used for if statements.
See https://google.github.io/styleguide/javaguide.html
If particular https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style

```
if (value) {
   stmt;
} else {
   stmt;
}
```

Reply:
done